src/Microsoft.DotNet.Cli.Utils/BuiltInCommand.cs

```diff
@@ -52,6 +54,11 @@ public CommandResult Execute()
                     // Reset the Reporters to the new Console Out and Error.
                     Reporter.Reset();
 
+                    if (!string.IsNullOrEmpty(_workingDirectory))
+                    {
+                        Directory.SetCurrentDirectory(_workingDirectory);
```

Comment:
Actually thinking more on this, I don't think this approach would work.  Why would a variable like `temp-out` mean that should be the working directory for `csc`?  Shouldn't it be named something else?  Like `project-directory` or `compiler-working-directory`?